chemprop/schedulers.py

```diff
@@ -31,97 +36,30 @@ class NoamLR(LRScheduler):
-        max_lrs: ArrayLike,
-        final_lrs: ArrayLike,
-    ):
-        self.num_lrs = len(optimizer.param_groups)
-        warmup_epochs = np.atleast_1d(warmup_epochs)
-        init_lrs = np.atleast_1d(init_lrs)
-        max_lrs = np.atleast_1d(max_lrs)
-        self.final_lrs = np.atleast_1d(final_lrs)
-
-        if not (
-            self.num_lrs
-            == len(warmup_epochs)
-            == len(init_lrs)
-            == len(max_lrs)
-            == len(self.final_lrs)
-        ):
-            raise ValueError(
-                "Number of param groups must match number of: "
-                "'warmup_epochs', 'init_lr', 'max_lr', 'final_lr'! "
-                f"got: {len(self.optimizer.param_groups)} param groups, "
-                f"{len(init_lrs)} init_lr, "
-                f"{len(max_lrs)} max_lr, "
-                f"{len(self.final_lrs)} final_lr"
-            )
-
-        self.current_step = 0
-        self.lrs = init_lrs
-
-        warmup_steps = (warmup_epochs * steps_per_epoch).astype(int)
-        total_steps = total_epochs * steps_per_epoch
-        cooldown_steps = total_steps - warmup_steps
-
-        deltas = (max_lrs - init_lrs) / warmup_steps
-        gammas = (self.final_lrs / max_lrs) ** (1 / cooldown_steps)
-
-        self.scheds = []
-        for i in range(self.num_lrs):
-            warmup = init_lrs[i] + np.arange(warmup_steps[i]) * deltas[i]
-            cooldown = max_lrs[i] * (gammas[i] ** np.arange(cooldown_steps[i]))
-            self.scheds.append(np.concatenate((warmup, cooldown)))
-        self.scheds = np.array(self.scheds)
-
-        super(NoamLR, self).__init__(optimizer)
-
-    def __len__(self) -> int:
-        """the number of steps in the learning rate schedule"""
-        return self.scheds.shape[1]
-
-    def get_lr(self) -> np.ndarray:
-        """Get a list of the current learning rates"""
-        return self.lrs
-
-    def step(self, step: int | None = None):
-        """Step the learning rate
-
-        Parameters
-        ----------
-        step : int | None, default=None
-            What step to set the learning rate to. If ``None``, use ``self.current_step + 1``.
-        """
-        self.current_step = step if step is not None else self.current_step + 1
-
-        for i in range(self.num_lrs):
-            if self.current_step < len(self):
-                self.lrs[i] = self.scheds[i][self.current_step]
-            else:
-                self.lrs[i] = self.final_lrs[i]
-
-            self.optimizer.param_groups[i]["lr"] = self.lrs[i]
+    def lr_lambda(step: int):
+        if step < warmup_steps:
+            warmup_factor = (max_lr - init_lr) / warmup_steps
+            return step * warmup_factor / init_lr + 1
+        elif step > warmup_steps + cooldown_steps:
+            return final_lr / init_lr
+        else:
+            cooldown_factor = (final_lr / max_lr) ** (1 / cooldown_steps)
+            return (max_lr * (cooldown_factor ** (step - warmup_steps))) / init_lr
```

Comment:
```suggestion
        elif warmup_steps <= step < warmup_steps + cooldown_steps:
            cooldown_factor = (final_lr / max_lr) ** (1 / cooldown_steps)
            return (max_lr * (cooldown_factor ** (step - warmup_steps))) / init_lr
        else:
            return final_lr / init_lr
```
IMO it makes more sense stylistically to have the cases ordered like `x < a`, `a <= x < b`, and `b < x`

Reply:
That is a good suggestion. Thanks!